src/resources/items/ItemsReport.tsx

```diff
@@ -1,12 +1,40 @@
-import { Typography } from '@mui/material'
-import React from 'react'
-import { Datagrid, List, TextField, type ListProps } from 'react-admin'
+import { Box, Button, Card, CardContent, Typography } from '@mui/material'
+import React, { useState } from 'react'
+import {
+  Datagrid,
+  List,
+  Show,
+  SimpleShowLayout,
+  TextField,
+  useListContext,
+  type ListProps
+} from 'react-admin'
 import SourceField from '../../components/SourceField'
 import * as constants from '../../constants'
+import { DateTime } from 'luxon'
+import { Download } from '@mui/icons-material'
+import Printable from '../../components/Printable'
 
-export default function ItemsReport(
-  props: Omit<ListProps, 'children'>
-): React.ReactElement {
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>
+
```

Comment:
the previous two declarations seem generic. Should they go in a central place, to encourage their re-use?

Reply:
Ok i will move to types.d.ts file